hpy/devel/src/runtime/ctx_call.c

```diff
@@ -36,3 +39,81 @@ ctx_CallTupleDict(HPyContext *ctx, HPy callable, HPy args, HPy kw)
     }
     return _py2h(obj);
 }
+
+#if PY_VERSION_HEX < 0x03090000
+#define PyObject_Vectorcall _PyObject_Vectorcall
+#endif
+
+_HPy_HIDDEN HPy
+ctx_Call(HPyContext *ctx, HPy h_callable, const HPy *h_args, size_t nargs, HPy h_kwnames)
+{
+    PyObject *kwnames;
+    size_t n_all_args;
+
+    if (HPy_IsNull(h_kwnames)) {
+        kwnames = NULL;
+        n_all_args = nargs;
+    } else {
+        kwnames = _h2py(h_kwnames);
+        assert(kwnames != NULL);
+        assert(PyTuple_Check(kwnames));
```

Comment:
does this need to be an `assert()`, or it should be a more proper check?
What happens if the user passes a non-tuple as kwnames?
Maybe it's good enough to have an `assert()` here, but then we should have a more proper check in debug mode